infra/README.md

```diff
@@ -0,0 +1,177 @@
+
+- Reads the current state of any already-existing remote objects to make sure that the Terraform state is up-to-date.
+- Compares the current configuration to the prior state and noting any differences.
+- Proposes a set of change actions that should, if applied, make the remote objects match the configuration.
+
+&nbsp;&nbsp;The `terraform apply` command executes the actions proposed in a Terraform plan.
+
+<span style="color:red"> ---- DANGER ----</span>\
+In a non-development environment comment out a resource to destroy, destroy should only be used as a way to cleanup a development environment. e.g. a developers workspace after they are done with it.
+
+&nbsp;&nbsp; The `terraform destroy` command is a convenient way to destroy all remote objects managed by a particular Terraform configuration.\
+
+For more information about terraform commands follow the link below:
+
+- [Basic CLI Features](https://www.terraform.io/cli/commands)
+
+## Terraform Backend
+
+### infra/bootstrap/account
+
+1. Rename bootstrap/account, account directory to the name of the aws account alias or account id where this infrastructure will be hosted.
+2. Customize the variables in locals{} at the top of main.tf to match the desired deployment setup.
+3. Open a terminal and cd into the infra/bootstrap/account directory and run the following commands:
+    - terraform init
+    - terraform plan
+    - terrafrom apply
+4. Uncomment out the backend "s3" {} block, fill in the appropriate information from outputs and repeate step `3.` to switch from local to remote backend.
+
+``` tf
+  backend "s3" {
+    bucket         = "AWS_ACCOUNT_ID-AWS_REGION-tf-state"
+    key            = "terraform/backend/terraform.tfstate"
+    region         = "REGION_OF_BUCKET"
+    encrypt        = "true"
+    dynamodb_table = "tf_state_locks"
+  }
+```
+5. Once these steps are complete, this should not need to be touched again, application infrastructure is managed under its envs/environment as described below.
+
+Note: For subsequent accounts if using a multi-account setup, copy the entire account directory and repeat the previous steps for each account.
+
+### infra/envs/environment
+
+&nbsp;&nbsp;To get started with an environment copy the backend configuration created in the above instructions and copy into the terraform { backend "s3" {} } block to setup the remote backend for the environment. This is where all of the infrastructure for the application will be managed. 
+
+### Multi vs Single Cloud Account.
+
+&nbsp;&nbsp;In a multi-cloud account, multi-environment setup the relationship between the bootstrap/account(s) and envs/environement(s) should be 1:1. In a single-cloud account, multi-environment setup ensure that the backend "s3" { key = path/to/terraform.tfstate} is unique for the backend, as well as each environment.
```

Comment:
```suggestion
&nbsp;&nbsp;In a simpler single cloud account setup, there is one cloud account that contains the resources created for managing terraform itself, as well as the resources created for each environment.

<insert single cloud diagram here>

In a multi-cloud account, multi-environment setup, the relationship between the bootstrap/account(s) and envs/environement(s) should be 1:1. In a single-cloud account, multi-environment setup ensure that the backend "s3" { key = path/to/terraform.tfstate} is unique for the backend, as well as each environment.

<insert multi-cloud diagram here>
```
I didn't make suggestions for your existing paragraph because I have the following questions:

> In a multi-cloud account, multi-environment setup the relationship between the bootstrap/account(s) and envs/environement(s) should be 1:1.

Does it need to be 1:1 between the bootstrap/accounts and the environments? Couldn't you have one bootstrap account for staging and one bootstrap account for test and prod?

> In a single-cloud account, multi-environment setup ensure that the backend "s3" { key = path/to/terraform.tfstate} is unique for the backend, as well as each environment.

Is this not necessary for multip-cloud, multip-environment setups? If so, can this just be simplified down to all `s3.keys` need to be unique?

Reply:
The keys only need to be unique if they share the same account and are not workspaces. Workspaces appends to the path in the key automatically. Alternative to unique keys would be to create multiple buckets in the single infra/account . We should probably discuss this more.